src/parsers/QASM3Parser.cpp

```diff
@@ -0,0 +1,807 @@
+    switch (condition->op) {
+    case BinaryExpression::Op::LessThan:
+      comparisonKind = qc::ComparisonKind::Lt;
+      break;
+    case BinaryExpression::Op::LessThanOrEqual:
+      comparisonKind = qc::ComparisonKind::Leq;
+      break;
+    case BinaryExpression::Op::GreaterThan:
+      comparisonKind = qc::ComparisonKind::Gt;
+      break;
+    case BinaryExpression::Op::GreaterThanOrEqual:
+      comparisonKind = qc::ComparisonKind::Geq;
+      break;
+    case BinaryExpression::Op::Equal:
+      comparisonKind = qc::ComparisonKind::Eq;
+      break;
+    case BinaryExpression::Op::NotEqual:
+      comparisonKind = qc::ComparisonKind::Neq;
+      break;
+    default:
+      error("Unsupported comparison operator.", ifStatement->debugInfo);
+    }
+
+    const auto lhs =
+        std::dynamic_pointer_cast<IdentifierExpression>(condition->lhs);
+    const auto rhs = std::dynamic_pointer_cast<Constant>(condition->rhs);
+
+    if (lhs == nullptr) {
+      error("Only classical registers are supported in conditions.",
+            ifStatement->debugInfo);
+    }
+    if (rhs == nullptr) {
+      error("Can only compare to constants.", ifStatement->debugInfo);
+    }
+
+    const auto creg = qc->getCregs().find(lhs->identifier);
+    if (creg == qc->getCregs().end()) {
+      error("Usage of unknown or invalid identifier '" + lhs->identifier +
+                "' in condition.",
+            ifStatement->debugInfo);
+    }
+
+    // translate statements in then block
+    auto thenOps = std::make_unique<qc::CompoundOperation>(qc->getNqubits());
+    for (const auto& statement : ifStatement->thenStatements) {
+      auto gateCall = std::dynamic_pointer_cast<GateCallStatement>(statement);
+      if (gateCall == nullptr) {
+        error("Only gate calls are supported in if statements.",
+              statement->debugInfo);
+      }
+      auto qregs = qc->getQregs();
+
+      auto op =
+          evaluateGateCall(gateCall, gateCall->identifier, gateCall->arguments,
+                           gateCall->operands, qregs);
+
+      thenOps->emplace_back(std::move(op));
+    }
+
+    if (!ifStatement->elseStatements.empty()) {
+      error("Else statements are not supported yet.", ifStatement->debugInfo);
+    }
+
+    std::unique_ptr<qc::Operation> thenOp = std::move(thenOps);
+
+    qc->emplace_back(std::make_unique<qc::ClassicControlledOperation>(
+        thenOp, creg->second, rhs->getUInt(), comparisonKind));
+  }
+
+  [[nodiscard]] std::unique_ptr<qc::Operation>
+  getBarrierOp(const std::shared_ptr<BarrierStatement>& barrierStatement,
+               const qc::QuantumRegisterMap& qregs) const {
+    std::vector<qc::Qubit> qubits{};
+    for (const auto& gate : barrierStatement->gates) {
+      translateGateOperand(gate, qubits, qregs, barrierStatement->debugInfo);
+    }
+
+    return std::make_unique<qc::NonUnitaryOperation>(qc->getNqubits(), qubits,
+                                                     qc::Barrier);
```

Comment:
`Barrier` is no longer a `NonUnitaryOperation` but has been promoted to a `StandardOperation` a while back.
Should be an easy fix though.
```suggestion
    return std::make_unique<qc::StandardOperation>(qc->getNqubits(), qubits,
                                                     qc::Barrier);
```